examples/source_examples/sql_server/connector.py

```diff
@@ -0,0 +1,220 @@
+# - configuration: a dictionary that holds the configuration settings for the connector.
+def schema(configuration: dict):
+    return [
+        {
+            "table": "employee_details",  # Name of the table in the destination.
+            "primary_key": ["employee_id"],  # Primary key column(s) for the table.
+            "columns": {  # Define the columns and their data types.
+                "employee_id": "INT",  # Integer column for the employee_id.
+                "first_name": "STRING",  # String column for the first name.
+                "last_name": "STRING",  # String column for the last name.
+                "hire_date": "NAIVE_DATE",  # NAIVE_DATE column for the hire_date.
+                "salary": "LONG",  # Integer column for the salary.
+            },
+        }
+    ]
+
+
+
+# Establishes a connection to a database using pyodbc.
+def connect_to_database(configuration):
+    # Define connection parameters
+    connection_string = (
+        "DRIVER=" + configuration.get('driver') + ";"
+        "SERVER=" + configuration.get('server') + ";"
+        "DATABASE=" + configuration.get('database') + ";"
+        "UID=" + configuration.get('user') + ";"
+        "PWD=" + configuration.get('password') + ";"
+    )
+
+    try:
+        # Establish connection
+        conn = pyodbc.connect(connection_string)
+        print("Connection established!")
+        return conn
+    except pyodbc.Error as e:
+        print(f"Error connecting to database: {e}")
+        return None
+
+
+# Converting Date to String
+def dt2str(incoming: datetime) -> str:
+    return incoming.strftime(TIMESTAMP_FORMAT)
+
+# Add mock data to your database, for illustration purposes
+def setup_db(configuration):
+    create_table_sql = """
+        CREATE TABLE employee_details (
+            employee_id INT IDENTITY(1,1) PRIMARY KEY, -- Auto-incrementing primary key
+            first_name NVARCHAR(50) NOT NULL,          -- Employee's first name
+            last_name NVARCHAR(50) NOT NULL,           -- Employee's last name
+            hire_date DATE NOT NULL,                   -- Date of hire
+            salary INT NOT NULL                        -- Salary
+        );
+        """
+
+    insert_data_sql = """
+        INSERT INTO employee_details (first_name, last_name, hire_date, salary)
+        VALUES 
+            ('John', 'Doe', '2020-05-15', 55000),
+            ('Jane', 'Smith', '2018-03-22', 62000),
+            ('Alice', 'Johnson', '2019-07-30', 58000),
+            ('Bob', 'Brown', '2021-11-01', 54000),
+            ('Charlie', 'Taylor', '2017-06-10', 67000),
+            ('Diana', 'Wilson', '2022-01-20', 51000),
+            ('Eve', 'Martin', '2015-12-15', 75000),
+            ('Frank', 'Moore', '2023-04-05', 52000),
+            ('Grace', 'Hall', '2020-09-14', 60000),
+            ('Hank', 'Lee', '2021-03-18', 53000);
+        """
+
+    try:
+        # Establish connection
+        conn = connect_to_database(configuration)
```

Comment:
Could we reuse the connection made in update method?

Reply:
this is optional method, let's keep it separate, we need it only to demonstrate dummy data.